src/apps/chifra/internal/scrape/pause.go

```diff
@@ -7,21 +7,32 @@ import (
 	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/rpcClient"
 )
 
-func (opts *ScrapeOptions) Pause(progressThen *rpcClient.MetaData) {
-	// we always pause at least a quarter of a second to allow the node to 'rest'
-	time.Sleep(250 * time.Millisecond)
+const (
+	minimumSleep = time.Second / 4
+)
+
+func (opts *ScrapeOptions) Pause(progressThen *rpcClient.MetaData) <-chan time.Time {
+	sleep := opts.computePause(progressThen)
+	return time.After(sleep)
+}
+
+func (opts *ScrapeOptions) computePause(progressThen *rpcClient.MetaData) time.Duration {
+	var sleep time.Duration
+
 	isDefaultSleep := opts.Sleep >= 13 && opts.Sleep <= 14
 	distanceFromHead := progressThen.Latest - progressThen.Staging
 	shouldSleep := !isDefaultSleep || distanceFromHead <= (2*opts.Settings.Unripe_dist)
 	if shouldSleep {
-		sleep := opts.Sleep
-		if sleep > 1 {
-			logger.Log(logger.Info, "Sleeping for", sleep, "seconds -", distanceFromHead, "away from head.")
-		}
-		halfSecs := (sleep * 2) - 1 // we already slept one quarter of a second
```

Comment:
This was here for a reason. In a process completely outside of this sleep loop, it's possible for some external code to signal that the scraper should (a) restart, (b) quit, (c) pause. If the user, for example, makes `Sleep` = 1000, then the scraper won't respond to those signals for 1000 seconds. If we come back alive every half second, the response is "snappy".